examples/go/tagging/gotag-fn/handler.go

```diff
@@ -0,0 +1,177 @@
+package function
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+
+	handler "github.com/openfaas-incubator/go-function-sdk"
+	"github.com/pelletier/go-toml"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vapi/tags"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+// tagInfo is part of the toml vcconfig file
+type tagInfo struct {
+	Action string
+	Urn    string
+}
+
+// VCenter is part of the toml vcconfig file
+type vCenterAuth struct {
+	Password string
+	Server   string
+	User     string
+}
+
+// Config represents the toml vcconfig file
+type vcConfig struct {
+	Tag     tagInfo
+	VCenter vCenterAuth
+}
+
+// The request from vmware-event-router.
+type dataObj struct {
+	Data nest1 `json:"data"`
+}
+type nest1 struct {
+	Vm nest2 `json:"Vm"`
+}
+type nest2 struct {
+	Vm vmInfo `json:"Vm"`
+}
+type vmInfo struct {
+	Value string `json:"Value"`
+	Type  string `json:"Type"`
+}
+
+// Handle a function invocation
+func Handle(req handler.Request) (handler.Response, error) {
+	ctx := context.Background()
+
+	// Retrieve vCenter and tag info from the config file.
+	vcc := loadTomlCfg()
+
+	// Incoming request object.
+	var mor dataObj
+	err := json.Unmarshal(req.Body, &mor)
+	if err != nil {
+		return handler.Response{
+			Body:       nil,
+			StatusCode: http.StatusBadRequest,
+		}, err
+	}
+
+	if mor.Data.Vm.Vm.Value == "" || mor.Data.Vm.Vm.Type == "" {
+		return handler.Response{
+			Body:       nil,
+			StatusCode: http.StatusBadRequest,
+		}, errors.New("managed Reference Object information must not be empty")
+	}
+
+	// Fill information in the request into a govmomi type.
+	var moRef types.ManagedObjectReference
+	moRef.Value = mor.Data.Vm.Vm.Value
+	moRef.Type = mor.Data.Vm.Vm.Type
+
+	// Connect to vcenter with govmomi client
+	log.Println("Connecting to vCenter")
+	clients := connect(ctx, vcc.VCenter)
```

Comment:
you're using a global var but closing the connection in each fn iteration. either make this a local variable then or persist the connection across invocation. also you don't check for errors when constructing the client. even though you log some errors to openfaas, it would be better to check for errors and return early.

Reply:
I have created a global var to persist the connection. Also, error is being checked.